esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6118,7 +6130,329 @@ bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &r
             userDesc.setown(createUserDescriptor());
             userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
         }
-        getFileAccess(context, userDesc, accessType, req, resp);
+        getFileAccess(context, userDesc, accessType, requestBase, resp.updateAccessInfo());
+    }
+    catch (IException *e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+bool CWsDfuEx::getGroupNameByClusterNameAndType(IPropertyTree *envTree, const char *clusterName, ClusterType clusterType, StringBuffer &groupName)
+{
+    StringBuffer xpath;
+    if (clusterType == ThorLCRCluster)
+        xpath.set("Software/ThorCluster");
+    else if (clusterType == RoxieCluster)
+        xpath.set("Software/RoxieCluster");
+    else
+        xpath.set("Software/EclAgentProcess");
```

Comment:
Code like this should be in esp.
I suspect there is either existing functions that can get the info. you want or something should be added.
Hard code environment paths like this shouldn't be added outside of environment (or some funcs are in workunit.* that should prob. be in environment).
Take a look at IClusterInfo

Reply:
@wangkx - I think you opened a JIRA for this, i.e. to come back and move common environment/workunit type code out of esp/services and into common helpers (and/or check there isn't already existing helper code you should use) ?

Can you reference/link it in the PR and in this JIRA please for future reference?